deepchem/models/torch_models/cnn.py

```diff
@@ -0,0 +1,442 @@
+    weight_decay_penalty_type: str
+      the type of penalty to use for weight decay, either 'l1' or 'l2'
+    dropouts: list or float
+      the dropout probability to use for each layer.  The length of this list should equal len(layer_filters).
+      Alternatively this may be a single value instead of a list, in which case the same value is used for every layer
+    activation_fns: str or list
+      the torch activation function to apply to each layer. The length of this list should equal
+      len(layer_filters).  Alternatively this may be a single value instead of a list, in which case the
+      same value is used for every layer, 'relu' by default
+    pool_type: str
+      the type of pooling layer to use, either 'max' or 'average'
+    mode: str
+      Either 'classification' or 'regression'
+    n_classes: int
+      the number of classes to predict (only used in classification mode)
+    uncertainty: bool
+      if True, include extra outputs and loss terms to enable the uncertainty
+      in outputs to be predicted
+    residual: bool
+      if True, the model will be composed of pre-activation residual blocks instead
+      of a simple stack of convolutional layers.
+    padding: str, int or tuple
+      the padding to use for convolutional layers, either 'valid' or 'same'
+    """
+    self.mode = mode
+    self.n_classes = n_classes
+    self.n_tasks = n_tasks
+
+    self.model = CNNModule(n_tasks=n_tasks,
+                           n_features=n_features,
+                           dims=dims,
+                           layer_filters=layer_filters,
+                           kernel_size=kernel_size,
+                           strides=strides,
+                           weight_init_stddevs=weight_init_stddevs,
+                           bias_init_consts=bias_init_consts,
+                           dropouts=dropouts,
+                           activation_fns=activation_fns,
+                           pool_type=pool_type,
+                           mode=mode,
+                           n_classes=n_classes,
+                           uncertainty=uncertainty,
+                           residual=residual,
+                           padding=padding)
+
+    regularization_loss: Optional[Callable]
+
+    if weight_decay_penalty != 0:
+      weights = [layer.weight for layer in self.model.layers]
+      if weight_decay_penalty_type == 'l1':
+        regularization_loss = lambda: weight_decay_penalty * torch.sum(
+            torch.stack([torch.abs(w).sum() for w in weights]))
+      else:
+        regularization_loss = lambda: weight_decay_penalty * torch.sum(
+            torch.stack([torch.square(w).sum() for w in weights]))
+    else:
+      regularization_loss = None
+
+    loss: Union[L2Loss, Callable[[Any, Any, Any], Any]]
+
+    if uncertainty:
+
+      def loss(outputs, labels, weights):
+
+        diff = labels[0] - outputs[0]
+
+        return torch.mean(diff**2 / torch.exp(outputs[1]) + outputs[1])
```

Comment:
I don't think this should be double spaced

Reply:
Sure. I've removed line spacing within the function